app/src/full/java/io/homeassistant/companion/android/location/HighAccuracyLocationService.kt

```diff
@@ -163,7 +163,7 @@ class HighAccuracyLocationService : Service() {
     override fun onDestroy() {
         super.onDestroy()
 
-        fusedLocationProviderClient.removeLocationUpdates(getLocationUpdateIntent())
+        if (fusedLocationProviderClient != null) fusedLocationProviderClient.removeLocationUpdates(getLocationUpdateIntent())
```

Comment:
I don't think this will fix the crash because you are still referencing a lateinit variable before it was initiallized.  I think you'll need to make the declaration of this `private var fusedLocationProviderClient: FusedLocationProviderClient? = null` and update all access with a `?.`

Reply:
Good catch. Learned something again. Thanks. 
Now i implemeted a nullable object. But I could also have checked the following: 
```
      if (::fusedLocationProviderClient.isInitialized) fusedLocationProviderClient.removeLocationUpdates(getLocationUpdateIntent())
```
I don't know which way is the recommenced approach in kotlin.